cirq-google/cirq_google/engine/engine_client.py

```diff
@@ -1137,3 +1337,12 @@ def _date_or_time_to_filter_expr(param_name: str, param: Union[datetime.datetime
         f"type {type(param)}. Supported types: datetime.datetime and"
         f"datetime.date"
     )
+
+
+def _get_job_path_from_stream_request(request: quantum.QuantumRunStreamRequest) -> str:
+    if 'create_quantum_program_and_job' in request:
+        return request.create_quantum_program_and_job.quantum_job.name
+    elif 'create_quantum_job' in request:
+        return request.create_quantum_job.quantum_job.name
+    # 'get_quantum_result' in request
+    return request.get_quantum_result.parent
```

Comment:
This won't be true if a new request type is added later. Prefer to be specific and raise an exception if an unexpected request comes through so that we can detect that it wasn't correctly handled.